{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import Entrez\n",
    "import urllib\n",
    "import requests\n",
    "import re\n",
    "from bs4 import BeautifulSoup\n",
    "from Bio.Blast import NCBIXML\n",
    "from Bio.Blast import NCBIWWW\n",
    "from Bio import pairwise2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_NCBIresults(ProtAcc):\n",
    "    All_Results = []\n",
    "    for ProtAcc in All_Proteins:\n",
    "        Results = {}\n",
    "        Results['Protein_Name'] = ProtAcc\n",
    "        Entrez.email = 'adrajesh@ucsd.edu'\n",
    "        paramEutils = { 'usehistory':'Y' }        # Use Entrez search history to cache results\n",
    "        try:\n",
    "            eSearch = Entrez.esearch(db=\"protein\", term=ProtAcc, **paramEutils)\n",
    "            res = Entrez.read(eSearch)\n",
    "            ProtID = res['IdList'][0]\n",
    "        except IndexError:\n",
    "            print('WARNING: Protein ID unavailable for', ProtAcc)\n",
    "            Results['Protein_ID'] =None\n",
    "            Results['organism'] = None\n",
    "            Results['genus'] =None\n",
    "            Results['species'] = None\n",
    "            Results['Gene_ID'] = None\n",
    "            Results['Start'] = None\n",
    "            Results['Assembly_Refseq_ID'] = None\n",
    "            Results['Assembly_GeneBank_ID'] = None\n",
    "        else:\n",
    "            Results['Protein_ID'] =ProtID\n",
    "            handle = Entrez.efetch(db=\"protein\", id=ProtAcc, rettype=\"gp\", retmode=\"xml\")\n",
    "            record =Entrez.read(handle)\n",
    "            handle.close()\n",
    "            Results['organism'] = record[0]['GBSeq_organism']\n",
    "            Results['genus']=record[0]['GBSeq_organism'].split(' ')[0]\n",
    "            try:\n",
    "                eSearch2 = Entrez.esearch(db=\"gene\", term=ProtAcc, **paramEutils)\n",
    "                res2 = Entrez.read(eSearch2)\n",
    "                GeneID = res2['IdList'][0]\n",
    "            except IndexError:\n",
    "                handle = Entrez.efetch(db=\"protein\", id=ProtID, rettype=\"gp\", retmode=\"text\")\n",
    "                record =handle.read()\n",
    "                handle.close()\n",
    "                g = record[record.find(\"GeneID\")::]\n",
    "                GeneID = g[g.find(\":\")+1:g.find('\\\"')]\n",
    "                if GeneID == '':\n",
    "                    print('WARNING: Gene ID unavailable for', ProtAcc)\n",
    "                    Results['Gene_ID'] = None\n",
    "                    Results['organism'] = None\n",
    "                    Results['genus'] =None\n",
    "                    Results['Start'] = None\n",
    "                    Results['Assembly_Refseq_ID'] = None\n",
    "                    Results['Assembly_GeneBank_ID'] = None\n",
    "                else:\n",
    "                    handle = Entrez.efetch(db=\"gene\", id=GeneID, rettype=\"gb\", retmode=\"text\")\n",
    "                    record =handle.read()\n",
    "                    handle.close()\n",
    "                    Ann = record[record.find(\"\\nAnnotation\")+1:record.find(\")\\n\")]\n",
    "                    AID = Ann[Ann.find(\"N\"):Ann.find(\".\")+2]\n",
    "                    #print(AID)\n",
    "                    Start = int(Ann[Ann.find(\"(\")+1:Ann.find(\"..\")])\n",
    "                    Results['Gene_ID'] = GeneID\n",
    "                    Results['Start'] =Start\n",
    "                    Results['Assembly_Refseq_ID'] = AID\n",
    "                    handle = Entrez.efetch(db=\"nucleotide\", id=AID, rettype=\"gb\", retmode=\"xml\")\n",
    "                    record = Entrez.read(handle)\n",
    "                    handle.close()\n",
    "                    gbID = record[0]['GBSeq_contig']\n",
    "                    gbID = gbID[gbID.find(\"(\")+1:gbID.find(\":\")]\n",
    "                    Results['Assembly_GeneBank_ID'] = gbID\n",
    "                    handle = Entrez.efetch(db=\"nucleotide\", rettype=\"gb\", retmode=\"xml\", id=gbID)\n",
    "                    record = Entrez.read(handle)\n",
    "                    handle.close()\n",
    "                    genome = record[0]['GBSeq_sequence']\n",
    "                    Results['Sequence'] = genome\n",
    "            else:\n",
    "                handle = Entrez.efetch(db=\"gene\", id=GeneID, rettype=\"gb\", retmode=\"text\")\n",
    "                record =handle.read()\n",
    "                handle.close()\n",
    "                Ann = record[record.find(\"\\nAnnotation\")+1:record.find(\")\\n\")]\n",
    "                AID = Ann[Ann.find(\"N\"):Ann.find(\".\")+2]\n",
    "                Start = int(Ann[Ann.find(\"(\")+1:Ann.find(\"..\")])\n",
    "                Results['Gene_ID'] = GeneID\n",
    "                Results['Start'] =Start\n",
    "                Results['Assembly_Refseq_ID'] = AID\n",
    "                handle = Entrez.efetch(db=\"nucleotide\", id=AID, rettype=\"gb\", retmode=\"xml\")\n",
    "                record = Entrez.read(handle)\n",
    "                handle.close()\n",
    "                gbID = record[0]['GBSeq_contig']\n",
    "                gbID = gbID[gbID.find(\"(\")+1:gbID.find(\":\")]\n",
    "                Results['Assembly_GeneBank_ID'] = gbID\n",
    "                handle = Entrez.efetch(db=\"nucleotide\",id=gbID, rettype=\"fasta\", retmode=\"xml\")\n",
    "                record =Entrez.read(handle)\n",
    "                handle.close()\n",
    "                genome = record[0]['TSeq_sequence']\n",
    "               # genome = record[0]['GBSeq_sequence']\n",
    "                Results['Sequence'] = genome\n",
    "        All_Results.append(Results)\n",
    "    return All_Results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_PHASTERresults(Results):\n",
    "    for Result in Results:\n",
    "        if Result['Gene_ID']!=None:\n",
    "            RID = Result['Assembly_Refseq_ID']\n",
    "            start = Result['Start']\n",
    "            flag=0\n",
    "            header = {'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.9; rv:32.0) Gecko/20100101 Firefox/32.0',}\n",
    "            url = \"http://phaster.ca/phaster_api?acc=\"+RID\n",
    "            r  = requests.get(url, headers=header)\n",
    "            data = r.text\n",
    "            all_matches = re.findall(r'\\d{1,15}-\\d{1,15}',data)\n",
    "            for match in all_matches:\n",
    "                positions = match.split(\"-\")\n",
    "                if int(positions[0])<=start and int(positions[1])>=start:\n",
    "                    PhageROI = [int(positions[0]),int(positions[1])]\n",
    "                    print('found_region for',RID)\n",
    "                    Result['Phage_region'] = PhageROI\n",
    "                    print(Result['Phage_region'])\n",
    "                    flag = 1\n",
    "            if flag ==0:\n",
    "                print('something is wrong, No results found for', Result['Protein_Name'],RID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#search = All_Results[0]['Sequence'][1657836: 1716259]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "#result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", search, entrez_query  = 'Streptococcus[organism]',hitlist_size=10,descriptions=10, alignments = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "#with open(\"my_blast2.xml\", \"w\") as out_handle:\n",
    "#     out_handle.write(result_handle.read())\n",
    "#result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "#result_handle = open(\"my_blast.xml\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#res = result_handle.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "#result_handle = open(\"my_blast.xml\")\n",
    "#blast_record = NCBIXML.read(result_handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['application', 'version', 'date', 'reference', 'query', 'query_letters', 'database', 'database_sequences', 'database_letters', 'database_name', 'posted_date', 'num_letters_in_database', 'num_sequences_in_database', 'ka_params', 'gapped', 'ka_params_gap', 'matrix', 'gap_penalties', 'sc_match', 'sc_mismatch', 'num_hits', 'num_sequences', 'num_good_extends', 'num_seqs_better_e', 'hsps_no_gap', 'hsps_prelim_gapped', 'hsps_prelim_gapped_attemped', 'hsps_gapped', 'query_id', 'query_length', 'database_length', 'effective_hsp_length', 'effective_query_length', 'effective_database_length', 'effective_search_space', 'effective_search_space_used', 'frameshift', 'threshold', 'window_size', 'dropoff_1st_pass', 'gap_x_dropoff', 'gap_x_dropoff_final', 'gap_trigger', 'blast_cutoff', 'descriptions', 'alignments', 'multiple_alignment', 'filter', 'expect'])"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#blast_record.__dict__.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['title', 'hit_id', 'hit_def', 'length', 'hsps', 'accession'])"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#blast_record.alignments[0].__dict__.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['score', 'bits', 'expect', 'num_alignments', 'identities', 'positives', 'gaps', 'align_length', 'strand', 'frame', 'query', 'query_start', 'query_end', 'match', 'sbjct', 'sbjct_start', 'sbjct_end'])"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#blast_record.alignments[0].hsps[0].__dict__.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getAllAlignments(Result,blastfile):\n",
    "    query_start = Result['Phage_region'][0]-10000\n",
    "    query_end = Result['Phage_region'][1] +10000\n",
    "    if query_end>=len(Result['Sequence']):\n",
    "        print(\"warning, query to exceeds query length, clipping\")\n",
    "    search = Result['Sequence'][query_start:query_end]\n",
    "    result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", search, entrez_query  = Result['genus']+'[organism]',hitlist_size=10)\n",
    "    with open(blastfile, \"w\") as out_handle:\n",
    "        out_handle.write(result_handle.read())\n",
    "    result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getOptAlignment(blastfile):\n",
    "    result_handle = open(blastfile)\n",
    "    record = NCBIXML.read(result_handle)\n",
    "    result_handle.close()\n",
    "    for align in record.alignments:\n",
    "        total_query = 0\n",
    "        for i in range(4):\n",
    "            total_query+= align.hsps[i].align_length\n",
    "            #print(total_query)\n",
    "        total_query_range = total_query/record.query_length\n",
    "        print(total_query_range)\n",
    "        if total_query_range<=0.6:\n",
    "            Opt_alignment = align\n",
    "            print(Opt_alignment.title)\n",
    "            return Opt_alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getAtt(Alignment):\n",
    "    return_pair =[]\n",
    "    all_start_ends =[]\n",
    "    for hsp in Alignment.hsps:\n",
    "        start_end = {}\n",
    "        start_end['subject_start'] = hsp.sbjct_start\n",
    "        start_end['subject_end'] = hsp.sbjct_end\n",
    "        start_end['query_start'] = hsp.query_start\n",
    "        start_end['query_end'] = hsp.query_end\n",
    "        start_end['expect'] =hsp.expect\n",
    "        start_end['query'] = hsp.query\n",
    "        start_end['subject'] = hsp.sbjct\n",
    "        all_start_ends.append(start_end)\n",
    "        sorted_start_ends = sorted(all_start_ends, key=lambda k: k['subject_start']) \n",
    "    while(1):\n",
    "        del_indices = []\n",
    "        for i in range(len(sorted_start_ends)-1):\n",
    "            if sorted_start_ends[i]['subject_start']< sorted_start_ends[i+1]['subject_start'] and sorted_start_ends[i]['subject_end']>sorted_start_ends[i+1]['subject_end']:\n",
    "                del_indices.append(i+1)\n",
    "    #    print(del_indices)\n",
    "        for index in sorted(del_indices, reverse=True):\n",
    "            del sorted_start_ends[index]\n",
    "        if len(del_indices) ==0:\n",
    "            break\n",
    "    possible_overlaps = []\n",
    "    c=0\n",
    "    for i in range(len(sorted_start_ends)-1):        \n",
    "        if sorted_start_ends[i]['subject_end']> sorted_start_ends[i+1]['subject_start'] and sorted_start_ends[i]['subject_end']<sorted_start_ends[i+1]['subject_end']:\n",
    "            if sorted_start_ends[i]['expect'] ==0 and sorted_start_ends[i+1]['expect'] == 0 and sorted_start_ends[i]['subject_end']-sorted_start_ends[i+1]['subject_start']<40 and sorted_start_ends[i+1]['query_start']-sorted_start_ends[i]['query_start']>2000:\n",
    "                return_pair.append(sorted_start_ends[i])\n",
    "                return_pair.append(sorted_start_ends[i+1])\n",
    "                print('\\nquery_start difference', sorted_start_ends[i+1]['query_start']-sorted_start_ends[i]['query_start'])\n",
    "                c = c+1\n",
    "                if c>1:\n",
    "                    print('warning: more than one subject overlap region')\n",
    "    return return_pair"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_index(alignment):\n",
    "    matches = []\n",
    "    breaks =[]\n",
    "    all_breaks =[]\n",
    "    overlaps = []\n",
    "    breaks.append(0)\n",
    "    for i, (a, b) in enumerate(zip(alignment[0], alignment[1])):\n",
    "        if a == b:\n",
    "            matches.append(i)\n",
    "            \n",
    "    for i in range(len(matches)-1):\n",
    "        if matches[i+1]-matches[i]>1:\n",
    "            breaks.append(matches[i])\n",
    "            breaks.append(matches[i+1])\n",
    "    breaks.append(len(alignment[0])-1)\n",
    "    \n",
    "    for i in range(0,len(breaks),2):\n",
    "        if breaks[i]!= breaks[i+1]:\n",
    "            all_breaks.append((breaks[i],breaks[i+1]))\n",
    "    for br in all_breaks:\n",
    "        overlaps.append(alignment[0][br[0]:br[1]+1])\n",
    "    return all_breaks,overlaps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getattsites(optalign,search):\n",
    "    potential_attB_sites = []\n",
    "    potential_attP_sites = []\n",
    "    pair1 = getAtt(optalign)\n",
    "    if(len(pair1) ==0):\n",
    "        print('No Site Found')\n",
    "        return None,None,None,None,None\n",
    "    align_length = pair1[0]['subject_end']-pair1[1]['subject_start']+1\n",
    "    X = pair1[0]['query'][-align_length::].upper()\n",
    "    Y = pair1[1]['query'][0:align_length].upper()\n",
    "    alignments = pairwise2.align.localms(X,Y,3,-1,-3,-2)\n",
    "\n",
    "    # Use format_alignment method to format the alignments in the list\n",
    "    print('query overlap aligned:')\n",
    "    for i in alignments:\n",
    "        print(pairwise2.format_alignment(*i))\n",
    "    range_1_sbjct_end = pair1[0]['subject_end']\n",
    "    range_1_query_end = pair1[0]['query_end']\n",
    "    range_2_sbjct_start = pair1[1]['subject_start']\n",
    "    range_2_query_start = pair1[1]['query_start']\n",
    "    overlap_size = range_1_sbjct_end - range_2_sbjct_start + 1\n",
    "    attL_start = range_1_query_end - overlap_size - 25\n",
    "    attL_stop = range_1_query_end + 25\n",
    "    attR_start = range_2_query_start - 25-1\n",
    "    attR_stop = range_2_query_start + overlap_size + 25-1\n",
    "    attL_seq = search[attL_start:attL_stop].upper()\n",
    "    attR_seq = search[attR_start:attR_stop].upper()\n",
    "    reference_attB = pair1[0]['subject'][-overlap_size-25::]+pair1[1]['subject'][overlap_size:overlap_size+25]\n",
    "    reference_attB = reference_attB.upper()\n",
    "    print('reference attB: ',reference_attB)\n",
    "    print('attL: ',attL_seq)\n",
    "    print('attR: ',attR_seq)\n",
    "    inds,overlaps = get_index(alignments[0])\n",
    "    print('\\npossible overlaps: ',overlaps)\n",
    "    for ind,overlap in zip(inds,overlaps):\n",
    "        if len(overlap)<= 18:\n",
    "            attB = attL_seq[0:25+ind[0]][-25::]+overlap+attR_seq[25+ind[1]+1::][0:25]\n",
    "            attP = attR_seq[0:25+ind[0]][-25::]+overlap+attL_seq[25+ind[1]+1::][0:25]\n",
    "            potential_attB_sites.append(attB)\n",
    "            potential_attP_sites.append(attP)\n",
    "        else:\n",
    "            print('overlap more than 18bp, ignoring.')\n",
    "    max_score =0  \n",
    "    print('potential attB sites:',potential_attB_sites)\n",
    "    print('\\nchecking possible attB sites:\\n')\n",
    "    for i,attBsite in enumerate(potential_attB_sites):\n",
    "        alignments = pairwise2.align.localms(attBsite,reference_attB,1,0,-3,-2)\n",
    "        score = alignments[0][2] - len(overlaps[i])\n",
    "        if score>max_score:\n",
    "            max_score = score\n",
    "            correct_index = i\n",
    "        # Use format_alignment method to format the alignments in the list\n",
    "        for i in alignments:\n",
    "            print(pairwise2.format_alignment(*i))\n",
    "        print(\"Corrected score=\",score)\n",
    "        print('\\n')\n",
    "    if max_score>0:\n",
    "        print('Best Regions:\\n')\n",
    "        print('attB: ',potential_attB_sites[correct_index])\n",
    "        print('attP: ',potential_attP_sites[correct_index])\n",
    "        print('overlap: ',overlaps[correct_index])\n",
    "        return overlaps[correct_index],potential_attB_sites[correct_index], potential_attP_sites[correct_index],attL_seq,attR_seq\n",
    "    else:\n",
    "        print('all scores<0')\n",
    "        return None,None,None,None,None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getBlastResults(Results):\n",
    "    for Result in Results:\n",
    "        if 'Phage_region'in Result:\n",
    "            print('\\n\\n\\n******',Result['organism'])\n",
    "            print(Result['Protein_Name'])\n",
    "            if not os.path.isfile('BlastFiles/'+Result['Protein_Name']+'.xml'):\n",
    "                getAllAlignments(Result, 'BlastFiles/'+Result['Protein_Name']+'.xml')\n",
    "            optalign =getOptAlignment('BlastFiles/'+Result['Protein_Name']+'.xml')\n",
    "            if optalign!= None:\n",
    "                search = Result['Sequence'][Result['Phage_region'][0]-10000:Result['Phage_region'][1] +10000]\n",
    "                o,b,p,l,r = getattsites(optalign,search)\n",
    "                if o!=None:\n",
    "                    Result['overlap'] = o\n",
    "                    Result['attB_site']=b\n",
    "                    Result['attP_site']=p\n",
    "                    Result['attL_site']=l\n",
    "                    Result['attR_site']=r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "All_Proteins = ['YP_353073','CBG73463','NP_268897','YP_002747001','BAF03598','BAG46462','YP_003251752','BAE05705','BAF67264','YP_003880342','YP_001886479','YP_005759947','YP_001376196','NP_470568','YP_006685721','YP_006538656','YP_189066','YP_002736920','FM864213','YP_006082695','YP_003445547','YP_004586821','YP_001089468','YP_005679179','YP_001384783','YP_001392519','YP_005869510','YP_001271396','YP_001646422','YP_002336631','YP_005549228','YP_706485','YP_002804732','YP_003472505']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Gene ID unavailable for BAF03598\n",
      "WARNING: Protein ID unavailable for FM864213\n"
     ]
    }
   ],
   "source": [
    "All_Results = get_NCBIresults(All_Proteins)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "found_region for NC_007493.2\n",
      "[1667836, 1706259]\n",
      "something is wrong, No results found for CBG73463 NC_013929.1\n",
      "found_region for NC_002737.2\n",
      "[526678, 570489]\n",
      "found_region for NC_012471.1\n",
      "[1729455, 1775766]\n",
      "found_region for NC_010805.1\n",
      "[1616527, 1665330]\n",
      "found_region for NC_013411.1\n",
      "[594415, 626069]\n",
      "found_region for NC_007168.1\n",
      "[2352533, 2397948]\n",
      "found_region for NC_009641.1\n",
      "[1098794, 1162158]\n",
      "found_region for NC_014498.1\n",
      "[1989782, 2034139]\n",
      "found_region for NC_010674.1\n",
      "[2326784, 2362434]\n",
      "found_region for NC_017353.1\n",
      "[887270, 936704]\n",
      "found_region for NC_009674.1\n",
      "[2983420, 3021377]\n",
      "found_region for NC_003212.1\n",
      "[1245851, 1297054]\n",
      "found_region for NC_018588.1\n",
      "[2381577, 2420764]\n",
      "found_region for NC_018221.1\n",
      "[2310353, 2378090]\n",
      "found_region for NC_002976.3\n",
      "[1557651, 1638109]\n",
      "found_region for NC_012466.1\n",
      "[1737927, 1786714]\n",
      "found_region for NC_017621.1\n",
      "[1166675, 1203771]\n",
      "found_region for NC_013853.1\n",
      "[390924, 446969]\n",
      "found_region for NC_015660.1\n",
      "[735196, 791797]\n",
      "found_region for NC_009089.1\n",
      "[3367045, 3442668]\n",
      "found_region for NC_017299.1\n",
      "[2687009, 2737939]\n",
      "found_region for NC_009697.1\n",
      "[2542387, 2595388]\n",
      "something is wrong, No results found for YP_001392519 NC_009699.1\n",
      "found_region for NC_017486.1\n",
      "[2145779, 2187692]\n",
      "found_region for NC_009513.1\n",
      "[860952, 878592]\n",
      "found_region for NC_010184.1\n",
      "[3633083, 3680422]\n",
      "found_region for NC_011658.1\n",
      "[571521, 628180]\n",
      "found_region for NC_017191.1\n",
      "[1172180, 1205688]\n",
      "found_region for NC_008268.1\n",
      "[7055617, 7072621]\n",
      "found_region for NC_012563.1\n",
      "[2563102, 2627075]\n",
      "found_region for NC_013893.1\n",
      "[2312214, 2349922]\n"
     ]
    }
   ],
   "source": [
    "get_PHASTERresults(All_Results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "****** Rhodobacter sphaeroides 2.4.1\n",
      "YP_353073\n",
      "1.0144121322082056\n",
      "1.0144121322082056\n",
      "1.0144292487547713\n",
      "1.0144121322082056\n",
      "1.0144292487547713\n",
      "0.5710764596135084\n",
      "gi|1523754079|gb|CP033434.1| Rhodobacter sphaeroides strain AB24 chromosome 1\n",
      "\n",
      "query_start difference 48470\n",
      "query overlap aligned:\n",
      "AAGAATGTGAAGGG\n",
      "|||||..|||||||\n",
      "AAGAAGATGAAGGG\n",
      "  Score=34\n",
      "\n",
      "reference attB:  GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "attL:  GAACTCCGCCGGGCCCATCTGGTCGAAGAATGTGAAGGGTATTTTACCCTTGTCGTTTCAGTGA\n",
      "attR:  TGGGGTCACAATACCAATCATGTTCAAGAAGATGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "\n",
      "possible overlaps:  ['AAGAA', 'TGAAGGG']\n",
      "potential attB sites: ['GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC', 'GCCGGGCCCATCTGGTCGAAGAATGTGAAGGGGCCCACCATCTGCCTCCGGGCCGAG']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC\n",
      "  |||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "1 GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC\n",
      "  Score=55\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      "1 GCCGGGCCCATCTGGTCGAAGAATGTGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "  |||||||||||||||||||||||..||||||||||||||||||||||||||||||||\n",
      "8 GCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "  Score=55\n",
      "\n",
      "Corrected score= 48.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC\n",
      "attP:  TGGGGTCACAATACCAATCATGTTCAAGAATGTGAAGGGTATTTTACCCTTGTCG\n",
      "overlap:  AAGAA\n",
      "\n",
      "\n",
      "\n",
      "****** Streptococcus phage 370.1\n",
      "NP_268897\n",
      "1.0962686684114025\n",
      "0.9787967591794519\n",
      "0.9785773612700005\n",
      "1.0420930560561659\n",
      "1.2756421306671264\n",
      "0.9937471595806365\n",
      "0.8965068718559496\n",
      "0.7356882042281111\n",
      "0.7356882042281111\n",
      "0.7224616445440442\n",
      "\n",
      "\n",
      "\n",
      "****** Streptococcus equi subsp. equi 4047\n",
      "YP_002747001\n",
      "1.2650691438826138\n",
      "1.265084224336837\n",
      "1.180995611587821\n",
      "0.5359141017327442\n",
      "gi|1654220791|emb|LR594033.1| Streptococcus equi subsp. zooepidemicus strain NCTC4675 genome assembly, chromosome: 1\n",
      "\n",
      "query_start difference 34139\n",
      "query overlap aligned:\n",
      "ATTTGAATTAG\n",
      "||||||||.||\n",
      "ATTTGAATAAG\n",
      "  Score=29\n",
      "\n",
      "reference attB:  CAAAGAGCGCCCAACGCGACCTGAAATTTGAATAAGACTGCTGCTTGTGTAAAGGCGATGA\n",
      "attL:  CAAAGAGCGCCCAACGCGACCTGAAATTTGAATTAGCGGTCAAATAATTTGTAATTCGTTT\n",
      "attR:  AAAATTACAAAGTTTTCAACCCTTGATTTGAATAAGACTGCTGCTTGTGTAAAGGCGATGA\n",
      "\n",
      "possible overlaps:  ['ATTTGAAT', 'AG']\n",
      "potential attB sites: ['CAAAGAGCGCCCAACGCGACCTGAAATTTGAATAAGACTGCTGCTTGTGTAAAGGCGA', 'CCCAACGCGACCTGAAATTTGAATTAGACTGCTGCTTGTGTAAAGGCGATGA']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 CAAAGAGCGCCCAACGCGACCTGAAATTTGAATAAGACTGCTGCTTGTGTAAAGGCGA\n",
      "  ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "1 CAAAGAGCGCCCAACGCGACCTGAAATTTGAATAAGACTGCTGCTTGTGTAAAGGCGA\n",
      "  Score=58\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      " 1 CCCAACGCGACCTGAAATTTGAATTAGACTGCTGCTTGTGTAAAGGCGATGA\n",
      "   ||||||||||||||||||||||||.|||||||||||||||||||||||||||\n",
      "10 CCCAACGCGACCTGAAATTTGAATAAGACTGCTGCTTGTGTAAAGGCGATGA\n",
      "  Score=51\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  CAAAGAGCGCCCAACGCGACCTGAAATTTGAATAAGACTGCTGCTTGTGTAAAGGCGA\n",
      "attP:  AAAATTACAAAGTTTTCAACCCTTGATTTGAATTAGCGGTCAAATAATTTGTAATTCG\n",
      "overlap:  ATTTGAAT\n",
      "\n",
      "\n",
      "\n",
      "****** Burkholderia multivorans ATCC 17616\n",
      "BAG46462\n",
      "1.0412482013865674\n",
      "1.0412482013865674\n",
      "0.3844018429428949\n",
      "gi|1061774371|gb|CP013466.1| Burkholderia multivorans strain MSMB1640WGS chromosome 2, complete sequence\n",
      "\n",
      "query_start difference 58799\n",
      "query overlap aligned:\n",
      "GGTCACGCTCGGCAACCCGAA\n",
      "||.||||||||||||.||.||\n",
      "GGACACGCTCGGCAATCCCAA\n",
      "  Score=51\n",
      "\n",
      "reference attB:  TACGGATGTTCGTCGCCGGCACGCTGGTCACGCTCGGGAACCCGAAGATCATGCTGTTCTATCTCGCGCTG\n",
      "attL:  TACGGATGTTCGTCGCCGGCACGCTGGTCACGCTCGGCAACCCGAACGAGAGTCAAAATACATTTGCGATG\n",
      "attR:  CCGAGTTGTCTGATAATATATTTTCGGACACGCTCGGCAATCCCAAGATCATGCTGTTCTATCTCGCGCTG\n",
      "\n",
      "possible overlaps:  ['GG', 'CACGCTCGGCAA', 'CC', 'AA']\n",
      "potential attB sites: ['TACGGATGTTCGTCGCCGGCACGCTGGACACGCTCGGCAATCCCAAGATCAT', 'GGATGTTCGTCGCCGGCACGCTGGTCACGCTCGGCAATCCCAAGATCATGCTGTTCTATCTC', 'CGGCACGCTGGTCACGCTCGGCAACCCCAAGATCATGCTGTTCTATCTCGCG', 'CACGCTGGTCACGCTCGGCAACCCGAAGATCATGCTGTTCTATCTCGCGCTG']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 TACGGATGTTCGTCGCCGGCACGCTGGACACGCTCGGCAATCCCAAGATCAT\n",
      "  |||||||||||||||||||||||||||.|||||||||.||.||.||||||||\n",
      "1 TACGGATGTTCGTCGCCGGCACGCTGGTCACGCTCGGGAACCCGAAGATCAT\n",
      "  Score=48\n",
      "\n",
      "Corrected score= 46.0\n",
      "\n",
      "\n",
      "1 GGATGTTCGTCGCCGGCACGCTGGTCACGCTCGGCAATCCCAAGATCATGCTGTTCTATCTC\n",
      "  ||||||||||||||||||||||||||||||||||.||.||.|||||||||||||||||||||\n",
      "4 GGATGTTCGTCGCCGGCACGCTGGTCACGCTCGGGAACCCGAAGATCATGCTGTTCTATCTC\n",
      "  Score=59\n",
      "\n",
      "Corrected score= 47.0\n",
      "\n",
      "\n",
      " 1 CGGCACGCTGGTCACGCTCGGCAACCCCAAGATCATGCTGTTCTATCTCGCG\n",
      "   |||||||||||||||||||||.|||||.||||||||||||||||||||||||\n",
      "17 CGGCACGCTGGTCACGCTCGGGAACCCGAAGATCATGCTGTTCTATCTCGCG\n",
      "  Score=50\n",
      "\n",
      "Corrected score= 48.0\n",
      "\n",
      "\n",
      " 1 CACGCTGGTCACGCTCGGCAACCCGAAGATCATGCTGTTCTATCTCGCGCTG\n",
      "   ||||||||||||||||||.|||||||||||||||||||||||||||||||||\n",
      "20 CACGCTGGTCACGCTCGGGAACCCGAAGATCATGCTGTTCTATCTCGCGCTG\n",
      "  Score=51\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  CACGCTGGTCACGCTCGGCAACCCGAAGATCATGCTGTTCTATCTCGCGCTG\n",
      "attP:  ATTTTCGGACACGCTCGGCAATCCCAACGAGAGTCAAAATACATTTGCGATG\n",
      "overlap:  AA\n",
      "\n",
      "\n",
      "\n",
      "****** Geobacillus sp. Y412MC61\n",
      "YP_003251752\n",
      "1.0085956557091416\n",
      "1.0085956557091416\n",
      "0.3064622294498006\n",
      "gi|922672232|gb|CP008934.1| Geobacillus stearothermophilus 10, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Staphylococcus haemolyticus JCSC1435\n",
      "BAE05705\n",
      "1.048199954138959\n",
      "0.28647863639837956\n",
      "gi|1663069024|gb|CP035541.1| Staphylococcus haemolyticus strain PK-01 chromosome, complete genome\n",
      "\n",
      "query_start difference 49105\n",
      "query overlap aligned:\n",
      "AA\n",
      "||\n",
      "AA\n",
      "  Score=6\n",
      "\n",
      "reference attB:  AGACAACTTTGGAGGCACATGCATAAACCACGGCTGTATACCCTCTAAAGTA\n",
      "attL:  AGATAACTATGGCGGCACGTGCATTAATGTTGAGTGAACAAACTTCCATAAT\n",
      "attR:  ACTATGGAAGTATGTACAGTCTTGCAACCACGGTTGTATCCCGTCTAAAGTA\n",
      "\n",
      "possible overlaps:  ['AA']\n",
      "potential attB sites: ['AGATAACTATGGCGGCACGTGCATTAACCACGGTTGTATCCCGTCTAAAGTA']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "AGATAACTATGGCGGCACGTGCATTAACCACGGTTGTATCCCGTCTAAAGTA\n",
      "|||.||||.|||.|||||.|||||.||||||||.|||||.||.|||||||||\n",
      "AGACAACTTTGGAGGCACATGCATAAACCACGGCTGTATACCCTCTAAAGTA\n",
      "  Score=44\n",
      "\n",
      "Corrected score= 42.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  AGATAACTATGGCGGCACGTGCATTAACCACGGTTGTATCCCGTCTAAAGTA\n",
      "attP:  ACTATGGAAGTATGTACAGTCTTGCAATGTTGAGTGAACAAACTTCCATAAT\n",
      "overlap:  AA\n",
      "\n",
      "\n",
      "\n",
      "****** Staphylococcus aureus subsp. aureus str. Newman\n",
      "BAF67264\n",
      "1.6025742526750155\n",
      "1.6025742526750155\n",
      "1.6025742526750155\n",
      "1.6025742526750155\n",
      "0.824636533755578\n",
      "0.824636533755578\n",
      "0.8555731490811381\n",
      "0.6306799097931961\n",
      "0.7843313660572909\n",
      "0.6268773091502328\n",
      "\n",
      "\n",
      "\n",
      "****** Streptococcus pneumoniae 670-6B\n",
      "YP_003880342\n",
      "1.1053809220442221\n",
      "1.1053809220442221\n",
      "0.7043678232360117\n",
      "0.6970492720294607\n",
      "0.6492844601208881\n",
      "0.39198533182093637\n",
      "gi|1489889596|dbj|AP018936.1| Streptococcus pneumoniae NU83127 DNA, complete genome\n",
      "\n",
      "query_start difference 50732\n",
      "query overlap aligned:\n",
      "ACCAACATCTCAATCAA\n",
      "|||.||||.||.|.|||\n",
      "ACCTACATTTCCACCAA\n",
      "  Score=35\n",
      "\n",
      "reference attB:  AAGCACGCTGATAATCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTTTAACCTTAGCTTTT\n",
      "attL:  AAGCACGCTGATAATCAGCAAGACCACCAACATCTCAATCAAGGATAGTAAAACTCTCACTCTTTTC\n",
      "attR:  TTGGAAAATATAAATAATTTTAGTAACCTACATTTCCACCAATGTAAAAGCTTTAACCTTAGCTTTT\n",
      "\n",
      "possible overlaps:  ['ACC', 'ACAT', 'TC', 'CAA']\n",
      "potential attB sites: ['AAGCACGCTGATAATCAGCAAGACCACCTACATTTCCACCAATGTAAAAGCTT', 'ACGCTGATAATCAGCAAGACCACCAACATTTCCACCAATGTAAAAGCTTTAACC', 'GATAATCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTTTAACCTTA', 'TCAGCAAGACCACCAACATCTCAATCAATGTAAAAGCTTTAACCTTAGCTTTT']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 AAGCACGCTGATAATCAGCAAGACCACCTACATTTCCACCAATGTAAAAGCTT\n",
      "  ||||||||||||||||||||||||||||.||||.|||||||||||||||||||\n",
      "1 AAGCACGCTGATAATCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTT\n",
      "  Score=51\n",
      "\n",
      "Corrected score= 48.0\n",
      "\n",
      "\n",
      "1 ACGCTGATAATCAGCAAGACCACCAACATTTCCACCAATGTAAAAGCTTTAACC\n",
      "  |||||||||||||||||||||||||||||.||||||||||||||||||||||||\n",
      "5 ACGCTGATAATCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTTTAACC\n",
      "  Score=53\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      " 1 GATAATCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTTTAACCTTA\n",
      "   ||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "10 GATAATCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTTTAACCTTA\n",
      "  Score=52\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      " 1 TCAGCAAGACCACCAACATCTCAATCAATGTAAAAGCTTTAACCTTAGCTTTT\n",
      "   ||||||||||||||||||||||.|.||||||||||||||||||||||||||||\n",
      "15 TCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTTTAACCTTAGCTTTT\n",
      "  Score=51\n",
      "\n",
      "Corrected score= 48.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  GATAATCAGCAAGACCACCAACATCTCCACCAATGTAAAAGCTTTAACCTTA\n",
      "attP:  ATAAATAATTTTAGTAACCTACATTTCAATCAAGGATAGTAAAACTCTCACT\n",
      "overlap:  TC\n",
      "\n",
      "\n",
      "\n",
      "****** Clostridium botulinum B str. Eklund 17B (NRP)\n",
      "YP_001886479\n",
      "1.1292003593890387\n",
      "1.1292003593890387\n",
      "0.5760467205750225\n",
      "gi|728867886|gb|CP006903.1| Clostridium botulinum 202F, complete genome\n",
      "\n",
      "query_start difference 36042\n",
      "query overlap aligned:\n",
      "AC\n",
      "||\n",
      "AC\n",
      "  Score=6\n",
      "\n",
      "reference attB:  ATTTTGCAGATTCCCAGATGCCCCTACAGAAAGAGGTACAAAACATTTATTA\n",
      "attL:  ATTTTGCAGATTCCCAGATGCCCCTACCCATAAAGTCTTATTAGTAAACATA\n",
      "attR:  TATGTTTACTAATAAGACGCTCTCAACAGAAAGAGGTACAAAACATTTATTG\n",
      "\n",
      "possible overlaps:  ['AC']\n",
      "potential attB sites: ['ATTTTGCAGATTCCCAGATGCCCCTACAGAAAGAGGTACAAAACATTTATTG']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 ATTTTGCAGATTCCCAGATGCCCCTACAGAAAGAGGTACAAAACATTTATT\n",
      "  |||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "1 ATTTTGCAGATTCCCAGATGCCCCTACAGAAAGAGGTACAAAACATTTATT\n",
      "  Score=51\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  ATTTTGCAGATTCCCAGATGCCCCTACAGAAAGAGGTACAAAACATTTATTG\n",
      "attP:  TATGTTTACTAATAAGACGCTCTCAACCCATAAAGTCTTATTAGTAAACATA\n",
      "overlap:  AC\n",
      "\n",
      "\n",
      "\n",
      "****** Staphylococcus lugdunensis N920143\n",
      "YP_005759947\n",
      "1.0067114093959733\n",
      "0.9933894057666273\n",
      "0.7846156061871705\n",
      "0.7868479419304664\n",
      "0.5275225393899242\n",
      "gi|1752979934|gb|CP041722.1| Staphylococcus lugdunensis strain SL13 chromosome, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Bacillus cytotoxicus NVH 391-98\n",
      "YP_001376196\n",
      "1.0481391376365237\n",
      "1.0384077850820437\n",
      "1.3315216453577652\n",
      "0.9575892472005108\n",
      "0.8657625480960022\n",
      "0.8884000207050055\n",
      "0.8884000207050055\n",
      "0.8884000207050055\n",
      "0.5425056507410666\n",
      "gi|1390185008|gb|CP024111.1| Bacillus cytotoxicus strain CH_4 chromosome, complete genome\n",
      "\n",
      "query_start difference 47979\n",
      "query overlap aligned:\n",
      "CCAGTTG\n",
      "|||||||\n",
      "CCAGTTG\n",
      "  Score=21\n",
      "\n",
      "reference attB:  ATACATTGTTGTTGTTTTTCCAGATCCAGTTGGTCCTGTAAATATAAGCAATCCATG\n",
      "attL:  ATACATTGTTGTTGTTTTTCCAGATCCAGTTGTTTTAGTAACATAAATACAACTCCG\n",
      "attR:  TAACGGTTGTATTTGTAGAACTTGACCAGTTGGTCCTGTAAATATAAGCAATCCATG\n",
      "\n",
      "possible overlaps:  ['CCAGTTG']\n",
      "potential attB sites: ['ATACATTGTTGTTGTTTTTCCAGATCCAGTTGGTCCTGTAAATATAAGCAATCCATG']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "ATACATTGTTGTTGTTTTTCCAGATCCAGTTGGTCCTGTAAATATAAGCAATCCATG\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "ATACATTGTTGTTGTTTTTCCAGATCCAGTTGGTCCTGTAAATATAAGCAATCCATG\n",
      "  Score=57\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  ATACATTGTTGTTGTTTTTCCAGATCCAGTTGGTCCTGTAAATATAAGCAATCCATG\n",
      "attP:  TAACGGTTGTATTTGTAGAACTTGACCAGTTGTTTTAGTAACATAAATACAACTCCG\n",
      "overlap:  CCAGTTG\n"
     ]
    }
   ],
   "source": [
    "getBlastResults(All_Results[0:13])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "****** Listeria monocytogenes SLCC2372\n",
      "YP_006685721\n",
      "1.0622265024414146\n",
      "1.0622265024414146\n",
      "0.8835555105006166\n",
      "0.9337523442647879\n",
      "0.8207376619865849\n",
      "1.020426782908409\n",
      "0.9851487657762684\n",
      "0.729366245966175\n",
      "0.729366245966175\n",
      "0.729366245966175\n",
      "\n",
      "\n",
      "\n",
      "****** Enterococcus faecalis D32\n",
      "YP_006538656\n",
      "1.0423994437922428\n",
      "0.7702793576256312\n",
      "0.7439164776548093\n",
      "0.596145297878888\n",
      "gi|1678198132|gb|CP030042.1| Enterococcus faecalis strain C25 chromosome, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Staphylococcus epidermidis RP62A\n",
      "YP_189066\n",
      "1.0457803261064325\n",
      "0.8709510442174839\n",
      "0.8985446654323199\n",
      "0.8838917756674431\n",
      "0.20579744768958172\n",
      "gi|27316888|gb|AE015929.1| Staphylococcus epidermidis ATCC 12228, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Streptococcus pneumoniae JJA\n",
      "YP_002736920\n",
      "1.2434762382426912\n",
      "0.5876255687847994\n",
      "gi|1112988118|gb|CP018136.1| Streptococcus pneumoniae strain SP49, complete genome\n",
      "\n",
      "query_start difference 12524\n",
      "\n",
      "query_start difference 63980\n",
      "warning: more than one subject overlap region\n",
      "query overlap aligned:\n",
      "TCGCTTTCAAATT\n",
      "|||||||||.|||\n",
      "TCGCTTTCACATT\n",
      "  Score=35\n",
      "\n",
      "reference attB:  CTTACTATGGCTTGGCTAAACAGCTTCGCTTTCACATTGTACAACATCTTAGCCGTGCTATGA\n",
      "attL:  CTTACTATGACTTGGCTAAACAGCTTCGCTTTCAAATTTCTAGGCTCAGGCTGAAACAGTCTC\n",
      "attR:  TCCCGAATGCTAAAATCGTTCTTGATCGCTTTCACATTGTACAACATCTTAGCCGTGCTATGA\n",
      "\n",
      "possible overlaps:  ['TCGCTTTCA', 'ATT']\n",
      "potential attB sites: ['CTTACTATGACTTGGCTAAACAGCTTCGCTTTCACATTGTACAACATCTTAGCCGTGCT', 'CTTGGCTAAACAGCTTCGCTTTCAAATTGTACAACATCTTAGCCGTGCTATGA']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 CTTACTATGACTTGGCTAAACAGCTTCGCTTTCACATTGTACAACATCTTAGCCGTGCT\n",
      "  |||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "1 CTTACTATGGCTTGGCTAAACAGCTTCGCTTTCACATTGTACAACATCTTAGCCGTGCT\n",
      "  Score=58\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      " 1 CTTGGCTAAACAGCTTCGCTTTCAAATTGTACAACATCTTAGCCGTGCTATGA\n",
      "   ||||||||||||||||||||||||.||||||||||||||||||||||||||||\n",
      "11 CTTGGCTAAACAGCTTCGCTTTCACATTGTACAACATCTTAGCCGTGCTATGA\n",
      "  Score=52\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  CTTACTATGACTTGGCTAAACAGCTTCGCTTTCACATTGTACAACATCTTAGCCGTGCT\n",
      "attP:  TCCCGAATGCTAAAATCGTTCTTGATCGCTTTCAAATTTCTAGGCTCAGGCTGAAACAG\n",
      "overlap:  TCGCTTTCA\n",
      "\n",
      "\n",
      "\n",
      "****** Streptococcus suis D12\n",
      "YP_006082695\n",
      "1.0922481434776516\n",
      "1.0427875858203728\n",
      "0.6115314557937509\n",
      "0.5350287235533138\n",
      "gi|1036166672|gb|KX077898.1| Streptococcus suis strain YY060816 composite mobile genetic element CMGEYY060816 mobile element, complete sequence\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Streptococcus mitis B6\n",
      "YP_003445547\n",
      "1.2165033861529357\n",
      "0.3669800775856401\n",
      "gi|1489891749|dbj|AP018937.1| Streptococcus pneumoniae HU-OH DNA, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Geobacillus thermoglucosidasius C56-YS93\n",
      "YP_004586821\n",
      "0.5019777809689169\n",
      "gi|311214252|gb|CP002293.1| Geobacillus sp. Y4.1MC1, complete genome\n",
      "\n",
      "query_start difference 55726\n",
      "query overlap aligned:\n",
      "TGC\n",
      "|||\n",
      "TGC\n",
      "  Score=9\n",
      "\n",
      "reference attB:  AAACGTTCCGTCTGTTTGTGTCAGCTGCGCGAAATTAATGACCGGATCGTTTG\n",
      "attL:  AAACGTTCCGTCTGTTTGTGTCAGCTGCCTAACATTAACTCTTATACAGGTTA\n",
      "attR:  AACCTGTTTTAAAGTTAACTTTACATGCGCGAAATTAATGACCGGATCGTTTG\n",
      "\n",
      "possible overlaps:  ['TGC']\n",
      "potential attB sites: ['AAACGTTCCGTCTGTTTGTGTCAGCTGCGCGAAATTAATGACCGGATCGTTTG']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "AAACGTTCCGTCTGTTTGTGTCAGCTGCGCGAAATTAATGACCGGATCGTTTG\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "AAACGTTCCGTCTGTTTGTGTCAGCTGCGCGAAATTAATGACCGGATCGTTTG\n",
      "  Score=53\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  AAACGTTCCGTCTGTTTGTGTCAGCTGCGCGAAATTAATGACCGGATCGTTTG\n",
      "attP:  AACCTGTTTTAAAGTTAACTTTACATGCCTAACATTAACTCTTATACAGGTTA\n",
      "overlap:  TGC\n",
      "\n",
      "\n",
      "\n",
      "****** Clostridioides difficile 630\n",
      "YP_001089468\n",
      "1.3789883187099339\n",
      "1.3789987764449976\n",
      "1.3789883187099339\n",
      "1.3789883187099339\n",
      "0.7875615699151878\n",
      "0.7120671804900495\n",
      "0.3058782928793282\n",
      "gi|777206503|emb|LN831032.1| Peptoclostridium difficile strain NCTC13307 genome assembly, plasmid: 3\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Clostridium botulinum H04402 065\n",
      "YP_005679179\n",
      "1.0580854363456929\n",
      "0.6245594247849993\n",
      "0.41610038065698574\n",
      "gi|1321438088|gb|CP014151.1| Clostridium botulinum strain BrDura chromosome, complete genome\n",
      "\n",
      "query_start difference 60504\n",
      "query overlap aligned:\n",
      "TTCATT\n",
      "||||||\n",
      "TTCATT\n",
      "  Score=18\n",
      "\n",
      "reference attB:  TACTTAATATATCCATAAGAGAAATTTCATTTCCTTCTTTGTCTACTCCTATAGGA\n",
      "attL:  TACTTAATATATCCATAAGAGAAATTTCATTACATCACATATGTTATACACCTACT\n",
      "attR:  TAGGTGTATATCATACCTAACGCAATTCATTTCCTTCTTTGTCTACCCCTATAGGA\n",
      "\n",
      "possible overlaps:  ['TTCATT']\n",
      "potential attB sites: ['TACTTAATATATCCATAAGAGAAATTTCATTTCCTTCTTTGTCTACCCCTATAGGA']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "TACTTAATATATCCATAAGAGAAATTTCATTTCCTTCTTTGTCTACCCCTATAGGA\n",
      "||||||||||||||||||||||||||||||||||||||||||||||.|||||||||\n",
      "TACTTAATATATCCATAAGAGAAATTTCATTTCCTTCTTTGTCTACTCCTATAGGA\n",
      "  Score=55\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  TACTTAATATATCCATAAGAGAAATTTCATTTCCTTCTTTGTCTACCCCTATAGGA\n",
      "attP:  TAGGTGTATATCATACCTAACGCAATTCATTACATCACATATGTTATACACCTACT\n",
      "overlap:  TTCATT\n",
      "\n",
      "\n",
      "\n",
      "****** Clostridium botulinum A str. ATCC 19397\n",
      "YP_001384783\n",
      "1.09487541266558\n",
      "0.5425816084711168\n",
      "gi|152930382|gb|CP000727.1| Clostridium botulinum A str. Hall, complete genome\n",
      "\n",
      "query_start difference 63935\n",
      "query overlap aligned:\n",
      "AATTTTTATCTATTGCG\n",
      "|||.|||.|.|||||||\n",
      "AATATTTTTGTATTGCG\n",
      "  Score=39\n",
      "\n",
      "reference attB:  TACTTTATATTCAATTATGTGTCGTAATTTTTTTGTATTGCGACGAAAAAACACCATAAAATTCTAA\n",
      "attL:  TACTTTATATTCAATTATGTGTCGTAATTTTTATCTATTGCGTAACTTCTTCTACACCTGTAATATC\n",
      "attR:  CCTAGATATATACTTATAGATACTAAATATTTTTGTATTGCGACGAAAAAACACCATAAAATTCTAA\n",
      "\n",
      "possible overlaps:  ['AAT', 'TTT', 'TATTGCG']\n",
      "potential attB sites: ['TACTTTATATTCAATTATGTGTCGTAATATTTTTGTATTGCGACGAAAAAACA', 'TTATATTCAATTATGTGTCGTAATTTTTTTGTATTGCGACGAAAAAACACCAT', 'TCAATTATGTGTCGTAATTTTTATCTATTGCGACGAAAAAACACCATAAAATTCTAA']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 TACTTTATATTCAATTATGTGTCGTAATATTTTTGTATTGCGACGAAAAAACA\n",
      "  ||||||||||||||||||||||||||||.||||||||||||||||||||||||\n",
      "1 TACTTTATATTCAATTATGTGTCGTAATTTTTTTGTATTGCGACGAAAAAACA\n",
      "  Score=52\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      "1 TTATATTCAATTATGTGTCGTAATTTTTTTGTATTGCGACGAAAAAACACCAT\n",
      "  |||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "5 TTATATTCAATTATGTGTCGTAATTTTTTTGTATTGCGACGAAAAAACACCAT\n",
      "  Score=53\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      " 1 TCAATTATGTGTCGTAATTTTTATCTATTGCGACGAAAAAACACCATAAAATTCTAA\n",
      "   ||||||||||||||||||||||.|.||||||||||||||||||||||||||||||||\n",
      "11 TCAATTATGTGTCGTAATTTTTTTGTATTGCGACGAAAAAACACCATAAAATTCTAA\n",
      "  Score=55\n",
      "\n",
      "Corrected score= 48.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  TTATATTCAATTATGTGTCGTAATTTTTTTGTATTGCGACGAAAAAACACCAT\n",
      "attP:  GATATATACTTATAGATACTAAATATTTATCTATTGCGTAACTTCTTCTACAC\n",
      "overlap:  TTT\n",
      "\n",
      "\n",
      "\n",
      "****** Lactococcus lactis subsp. lactis CV56\n",
      "YP_005869510\n",
      "1.3271687690791918\n",
      "0.7101416503803725\n",
      "0.7101416503803725\n",
      "0.5431977129197422\n",
      "gi|1173070269|gb|CP015906.1| Lactococcus lactis subsp. lactis strain UC77, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Lactobacillus reuteri DSM 20016\n",
      "YP_001271396\n",
      "1.2349628055260362\n",
      "1.0867162592986186\n",
      "1.2349628055260362\n",
      "1.2349628055260362\n",
      "0.6006907545164718\n",
      "0.5110255047821467\n",
      "gi|1700697728|gb|CP030089.1| Lactobacillus reuteri strain YSJL-12 chromosome, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Bacillus weihenstephanensis KBAB4\n",
      "YP_001646422\n",
      "1.0703752654479572\n",
      "0.4185835845498151\n",
      "gi|1182074459|gb|CP020743.1| Bacillus mycoides strain Gnyt1, complete genome\n",
      "\n",
      "query_start difference 50844\n",
      "query overlap aligned:\n",
      "ATTAGGTTGACGCCA\n",
      "|||||.||.||.|||\n",
      "ATTAGATTTACTCCA\n",
      "  Score=33\n",
      "\n",
      "reference attB:  GATGTCTTTTGTAGCCATTAGGCGCATGAGATTTACTCCATTAAGCCCTAAAGCATCATTCGTCG\n",
      "attL:  GATGTCTTTTGTAGCCATTAGGCGCATTAGGTTGACGCCAACAGGGTGATGACAAAGCTAATGAA\n",
      "attR:  GCTTTGTCGTCACCTTGTTGGCGTAATTAGATTTACTCCATTAAGCCCTAAAGCATCATTCGTCG\n",
      "\n",
      "possible overlaps:  ['ATTAG', 'TT', 'AC', 'CCA']\n",
      "potential attB sites: ['GATGTCTTTTGTAGCCATTAGGCGCATTAGATTTACTCCATTAAGCCCTAAAGCA', 'TTTTGTAGCCATTAGGCGCATTAGGTTTACTCCATTAAGCCCTAAAGCATCA', 'TGTAGCCATTAGGCGCATTAGGTTGACTCCATTAAGCCCTAAAGCATCATTC', 'AGCCATTAGGCGCATTAGGTTGACGCCATTAAGCCCTAAAGCATCATTCGTCG']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 GATGTCTTTTGTAGCCATTAGGCGCATTAGATTTACTCCATTAAGCCCTAAAGCA\n",
      "  |||||||||||||||||||||||||||.|||||||||||||||||||||||||||\n",
      "1 GATGTCTTTTGTAGCCATTAGGCGCATGAGATTTACTCCATTAAGCCCTAAAGCA\n",
      "  Score=54\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      "1 TTTTGTAGCCATTAGGCGCATTAGGTTTACTCCATTAAGCCCTAAAGCATCA\n",
      "  |||||||||||||||||||||.||.|||||||||||||||||||||||||||\n",
      "7 TTTTGTAGCCATTAGGCGCATGAGATTTACTCCATTAAGCCCTAAAGCATCA\n",
      "  Score=50\n",
      "\n",
      "Corrected score= 48.0\n",
      "\n",
      "\n",
      " 1 TGTAGCCATTAGGCGCATTAGGTTGACTCCATTAAGCCCTAAAGCATCATTC\n",
      "   ||||||||||||||||||.||.||.|||||||||||||||||||||||||||\n",
      "10 TGTAGCCATTAGGCGCATGAGATTTACTCCATTAAGCCCTAAAGCATCATTC\n",
      "  Score=49\n",
      "\n",
      "Corrected score= 47.0\n",
      "\n",
      "\n",
      " 1 AGCCATTAGGCGCATTAGGTTGACGCCATTAAGCCCTAAAGCATCATTCGTCG\n",
      "   |||||||||||||||.||.||.||.||||||||||||||||||||||||||||\n",
      "13 AGCCATTAGGCGCATGAGATTTACTCCATTAAGCCCTAAAGCATCATTCGTCG\n",
      "  Score=49\n",
      "\n",
      "Corrected score= 46.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  GATGTCTTTTGTAGCCATTAGGCGCATTAGATTTACTCCATTAAGCCCTAAAGCA\n",
      "attP:  GCTTTGTCGTCACCTTGTTGGCGTAATTAGGTTGACGCCAACAGGGTGATGACAA\n",
      "overlap:  ATTAG\n",
      "\n",
      "\n",
      "\n",
      "****** Bacillus cereus AH187\n",
      "YP_002336631\n",
      "1.2033420733377687\n",
      "0.6320849476251973\n",
      "0.6322675745835453\n",
      "0.5319662401022711\n",
      "gi|221237819|gb|CP000227.1| Bacillus cereus Q1, complete genome\n",
      "\n",
      "query_start difference 53411\n",
      "query overlap aligned:\n",
      "GAAGTGAATCAGT\n",
      "|||.|.||.||||\n",
      "GAATTAAACCAGT\n",
      "  Score=27\n",
      "\n",
      "reference attB:  GAAGTGTAATATGTTTGGATATGGGGAAGTGAATCAGTACAACCGCCACAGTACCCTCATGTC\n",
      "attL:  GAAGTGTAATATGTTTGGATATGGGGAAGTGAATCAGTTTAATACTCCACCATGTACACGCAG\n",
      "attR:  TTGGAATAATAGTGTATATGGTAGAGAATTAAACCAGTACAACCGCCACAGTACCCTCATGTC\n",
      "\n",
      "possible overlaps:  ['GAA', 'AA', 'CAGT']\n",
      "potential attB sites: ['GAAGTGTAATATGTTTGGATATGGGGAATTAAACCAGTACAACCGCCACAGTA', 'TAATATGTTTGGATATGGGGAAGTGAACCAGTACAACCGCCACAGTACCCTC', 'TATGTTTGGATATGGGGAAGTGAATCAGTACAACCGCCACAGTACCCTCATGTC']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 GAAGTGTAATATGTTTGGATATGGGGAATTAAACCAGTACAACCGCCACAGTA\n",
      "  ||||||||||||||||||||||||||||.|.||.|||||||||||||||||||\n",
      "1 GAAGTGTAATATGTTTGGATATGGGGAAGTGAATCAGTACAACCGCCACAGTA\n",
      "  Score=50\n",
      "\n",
      "Corrected score= 47.0\n",
      "\n",
      "\n",
      "1 TAATATGTTTGGATATGGGGAAGTGAACCAGTACAACCGCCACAGTACCCTC\n",
      "  |||||||||||||||||||||||||||.||||||||||||||||||||||||\n",
      "7 TAATATGTTTGGATATGGGGAAGTGAATCAGTACAACCGCCACAGTACCCTC\n",
      "  Score=51\n",
      "\n",
      "Corrected score= 49.0\n",
      "\n",
      "\n",
      " 1 TATGTTTGGATATGGGGAAGTGAATCAGTACAACCGCCACAGTACCCTCATGTC\n",
      "   ||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "10 TATGTTTGGATATGGGGAAGTGAATCAGTACAACCGCCACAGTACCCTCATGTC\n",
      "  Score=54\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  TATGTTTGGATATGGGGAAGTGAATCAGTACAACCGCCACAGTACCCTCATGTC\n",
      "attP:  TAGTGTATATGGTAGAGAATTAAACCAGTTTAATACTCCACCATGTACACGCAG\n",
      "overlap:  CAGT\n",
      "\n",
      "\n",
      "\n",
      "****** Bacillus amyloliquefaciens XH7\n",
      "YP_005549228\n",
      "1.0489833295955744\n",
      "1.0489833295955744\n",
      "1.0489833295955744\n",
      "1.0291545189504374\n",
      "0.7997121925693355\n",
      "0.48985198474994396\n",
      "gi|1551417581|gb|CP018902.1| Bacillus amyloliquefaciens strain HK1 chromosome, complete genome\n",
      "No Site Found\n",
      "\n",
      "\n",
      "\n",
      "****** Rhodococcus jostii RHA1\n",
      "YP_706485\n",
      "1.1498486650091881\n",
      "0.7142471084207113\n",
      "0.665819911360934\n",
      "0.650037833747703\n",
      "0.6698735271862502\n",
      "0.589936223111015\n",
      "gi|226237899|dbj|AP011115.1| Rhodococcus opacus B4 DNA, complete genome\n",
      "\n",
      "query_start difference 18080\n",
      "query overlap aligned:\n",
      "GAGTACTCGC\n",
      "||.|||.|.|\n",
      "GACTACGCAC\n",
      "  Score=18\n",
      "\n",
      "reference attB:  CGCGCAGAACGGTGCGGTGATCAGCGAGTACGCGCCGGGCACAACTCCCGCGAAACATCG\n",
      "attL:  CGCGCAGAACGGTGCGGTGATCAGTGAGTACTCGCTCCAGGTACATTAACACCATGGATC\n",
      "attR:  TATGTGGTGGTAATAGCGAGTAGGGGACTACGCACCGGGCACGACACCGGCGAAGCATCG\n",
      "\n",
      "possible overlaps:  ['GA', 'TAC']\n",
      "potential attB sites: ['CGCGCAGAACGGTGCGGTGATCAGTGACTACGCACCGGGCACGACACCGGCG', 'GCAGAACGGTGCGGTGATCAGTGAGTACGCACCGGGCACGACACCGGCGAAGC']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 CGCGCAGAACGGTGCGGTGATCAGTGACTACGCACCGGGCACGACACCGGCG\n",
      "  ||||||||||||||||||||||||.||.|||||.||||||||.||.||.|||\n",
      "1 CGCGCAGAACGGTGCGGTGATCAGCGAGTACGCGCCGGGCACAACTCCCGCG\n",
      "  Score=46\n",
      "\n",
      "Corrected score= 44.0\n",
      "\n",
      "\n",
      "1 GCAGAACGGTGCGGTGATCAGTGAGTACGCACCGGGCACGACACCGGCGAAGC\n",
      "  |||||||||||||||||||||.||||||||.||||||||.||.||.|||||.|\n",
      "4 GCAGAACGGTGCGGTGATCAGCGAGTACGCGCCGGGCACAACTCCCGCGAAAC\n",
      "  Score=47\n",
      "\n",
      "Corrected score= 44.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  CGCGCAGAACGGTGCGGTGATCAGTGACTACGCACCGGGCACGACACCGGCG\n",
      "attP:  TATGTGGTGGTAATAGCGAGTAGGGGAGTACTCGCTCCAGGTACATTAACAC\n",
      "overlap:  GA\n",
      "\n",
      "\n",
      "\n",
      "****** Clostridium botulinum A2 str. Kyoto\n",
      "YP_002804732\n",
      "1.0409060055017685\n",
      "1.0409060055017685\n",
      "0.9312398032700987\n",
      "0.686637371536089\n",
      "0.5912019339549617\n",
      "gi|1125848848|gb|CP013681.1| Clostridium botulinum strain 1169 chromosome, complete genome\n",
      "\n",
      "query_start difference 2228\n",
      "query overlap aligned:\n",
      "2 ACAC\n",
      "  ||||\n",
      "2 ACAC\n",
      "  Score=12\n",
      "\n",
      "reference attB:  CTTCATCACTGTTGCTGTCATATTTCACACTCTAACCTCCTTTCTAAAATGGAAC\n",
      "attL:  CTTCATCACTATCGCTGTCATATTTTACACCACTTATAAAATATTCTGGATTATC\n",
      "attR:  GTATTAAAGTTTTAAAAGAATTATCCACACTCTCACCTCCTTTCTAAAATGGAAC\n",
      "\n",
      "possible overlaps:  ['TACAC']\n",
      "potential attB sites: ['CTTCATCACTATCGCTGTCATATTTTACACTCTCACCTCCTTTCTAAAATGGAAC']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "CTTCATCACTATCGCTGTCATATTTTACACTCTCACCTCCTTTCTAAAATGGAAC\n",
      "||||||||||.|.||||||||||||.|||||||.|||||||||||||||||||||\n",
      "CTTCATCACTGTTGCTGTCATATTTCACACTCTAACCTCCTTTCTAAAATGGAAC\n",
      "  Score=51\n",
      "\n",
      "Corrected score= 46.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  CTTCATCACTATCGCTGTCATATTTTACACTCTCACCTCCTTTCTAAAATGGAAC\n",
      "attP:  GTATTAAAGTTTTAAAAGAATTATCTACACCACTTATAAAATATTCTGGATTATC\n",
      "overlap:  TACAC\n",
      "\n",
      "\n",
      "\n",
      "****** Staphylococcus lugdunensis HKU09-01\n",
      "YP_003472505\n",
      "1.014556040756914\n",
      "0.882667914327303\n",
      "1.1151833367990573\n",
      "1.0631801483329868\n",
      "0.7624592777431205\n",
      "0.7259305468912456\n",
      "0.4888403687530325\n",
      "gi|1339503361|gb|CP014023.2| Staphylococcus lugdunensis strain FDAARGOS_143, complete genome\n",
      "No Site Found\n"
     ]
    }
   ],
   "source": [
    "getBlastResults(All_Results[14::])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 540,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "****** Rhodobacter sphaeroides 2.4.1\n",
      "YP_353073\n",
      "1.0144121322082056\n",
      "1.0144121322082056\n",
      "1.0144292487547713\n",
      "1.0144121322082056\n",
      "1.0144292487547713\n",
      "0.5710764596135084\n",
      "gi|1523754079|gb|CP033434.1| Rhodobacter sphaeroides strain AB24 chromosome 1\n",
      "\n",
      "query_start difference 48470\n",
      "query overlap aligned:\n",
      "AAGAATGTGAAGGG\n",
      "|||||..|||||||\n",
      "AAGAAGATGAAGGG\n",
      "  Score=34\n",
      "\n",
      "reference attB:  GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "attL:  GAACTCCGCCGGGCCCATCTGGTCGAAGAATGTGAAGGGTATTTTACCCTTGTCGTTTCAGTGA\n",
      "attR:  TGGGGTCACAATACCAATCATGTTCAAGAAGATGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "\n",
      "possible overlaps:  ['AAGAA', 'TGAAGGG']\n",
      "potential attB sites: ['GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC', 'GCCGGGCCCATCTGGTCGAAGAATGTGAAGGGGCCCACCATCTGCCTCCGGGCCGAG']\n",
      "\n",
      "checking possible attB sites:\n",
      "\n",
      "1 GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC\n",
      "  |||||||||||||||||||||||||||||||||||||||||||||||||||||||\n",
      "1 GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC\n",
      "  Score=55\n",
      "\n",
      "Corrected score= 50.0\n",
      "\n",
      "\n",
      "1 GCCGGGCCCATCTGGTCGAAGAATGTGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "  |||||||||||||||||||||||..||||||||||||||||||||||||||||||||\n",
      "8 GCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTCCGGGCCGAG\n",
      "  Score=55\n",
      "\n",
      "Corrected score= 48.0\n",
      "\n",
      "\n",
      "Best Regions:\n",
      "\n",
      "attB:  GAACTCCGCCGGGCCCATCTGGTCGAAGAAGATGAAGGGGCCCACCATCTGCCTC\n",
      "attP:  TGGGGTCACAATACCAATCATGTTCAAGAATGTGAAGGGTATTTTACCCTTGTCG\n",
      "overlap:  AAGAA\n"
     ]
    }
   ],
   "source": [
    "getBlastResults([All_Results[0]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
